src/sentry/static/sentry/app/utils/discover/discoverQuery.tsx

```diff
@@ -120,12 +121,16 @@ class DiscoverQuery extends React.Component<Props, State> {
     return !isAPIPayloadSimilar(thisAPIPayload, otherAPIPayload);
   };
 
-  getRoute(keyTransactions, trendsType) {
+  getRoute(keyTransactions, trendsType, trendStats) {
     if (keyTransactions) {
       return 'key-transactions';
     }
     if (trendsType) {
-      return 'events-trends';
+      if (trendStats) {
+        return 'events-trends-stats';
+      } else {
+        return 'events-trends';
+      }
```

Comment:
Are we at a point where we want to have different query components yet?

Reply:
Hmm I think combined with your other comment it might be worth splitting into different query components.
But for now I think it'd be better to keep the response as-is and change the api response + components in a separate PR.

Reply:
I'm good with doing that work separately. I think having separate endpoints is beginning to make more sense.

Reply:
I'm leaning towards leaving them together still, I think. Trends is still a specialization of discoverQuery so it doesn't make a ton of sense to split them apart and have to track changes in discoverQuery in 'trendsQuery' or whatever we'd call it. Thoughts?

Reply:
If the response shape for trends is going to be generally consistent, a separate UI component would let you use better types for the response. Right now approximately half the DiscoverQuery component's code is related to trends. This indicates that we could have two simpler component that import shared functions for shared logic.

Reply:
Hmm that's a good point about how much of the file is trends. Shared functions would address my maintainability concerns 👍

Reply:
@markstory @k-fish so is this PR :+1:? I've tasked out splitting this up [here](https://getsentry.atlassian.net/browse/VIS-179)

Reply:
If there is a ticket for it then 👍